src/Agent.Sdk/IBaseContext.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Collections.Generic;
+
+namespace Agent.Sdk
+{
+    public interface IBaseContext : ITraceWriter
```

Comment:
Can we name this something more descriptive and add comments? Hard to tell from the name. Is it `IGitExecutionContext`?

Reply:
I can make it IBaseExecutionContext. I am not sure I want to limit it just to Git